### PR TITLE
Fixes #36371 - properly convert spaces in provider friendly name

### PR DESCRIPTION
**Pull request by evgeni**
includes #9701 as the tests are based on it

### CODE REVIEW

File: app/graphql/types/compute_resource.rb [hunk 1]
```diff
@@ -13,5 +13,9 @@ class ComputeResource < BaseObject
     has_many :compute_attributes, Types::ComputeAttribute
     has_many :hosts, Types::Host
     has_many :networks, Types::Networks::Union, resolver: Resolvers::ComputeResource::Networks
+
+    def provider_friendly_name
+      object.provider_friendly_name.tr(' ', '_')
```
Comment (evgeni) [inline app/graphql/types/compute_resource.rb:18]: I didn't figure out how to nicely write a test for this (without writing a full on query test).

**Comment by evgeni** [comment]
I'd argue that `provider_friendly_name` should not be an enum at all (and then the weird validation would not fail), but I also kinda didn't want to change the schema?

**Comment by evgeni** [comment]
@kamils-iRonin opinions on this solution vs making `providerFriendlyName` a simple `String`?

**Comment by kamils-iRonin** [comment]
> opinions on this solution vs making `providerFriendlyName` a simple `String`?

@evgeni making `providerFriendlyName` a `String` seems like a simpler and cleaner solution, and I cannot recall the reasons why it was implemented as an enum

**Comment by ekohl** [comment]
Perhaps for validation? Or you can introspect the available providers this way?

**Comment by evgeni** [comment]
For both I'd use the normal `provider` property, not the friendly name.

File: app/graphql/types/compute_resource.rb (hunk 1 above)
Comment (kamils-iRonin) [reply app/graphql/types/compute_resource.rb:18]: I think the existing query test is enough, wdyt?

https://github.com/theforeman/foreman/blob/f49802123759130ab64b54dc35ac76120bc64b0d/test/graphql/queries/compute_resource_query_test.rb#L81

**Comment by evgeni** [comment]
@kamils-iRonin I've opeend https://github.com/theforeman/foreman/pull/9703 with the alternative solution :)

**Comment by evgeni** [comment]
Which now has been merged, closing

**Pull request closed without merging**